cmd/kube-apiserver/app/preflight/checks.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package preflight
+
+import (
+	"k8s.io/kubernetes/cmd/kube-apiserver/app/options"
+	"net"
+	"net/url"
+	"time"
+	"fmt"
+	"errors"
+)
+
+var retryLimit int = 6
+var retryInterval time.Duration = 10
+
+type clock interface {
+	Sleep(time.Duration)
+}
+
+type realTimer struct{}
+func (realTimer) Sleep(d time.Duration) {time.Sleep(d)}
+
+var timer clock = new(realTimer)
```

Comment:
I don't understand the purpose of using underscore here. timer is actually used. Perhaps I'm unclear on the construct, but the point here is to allow mocking out sleeping for 10 second intervals (so the test suite won't take 4 minutes while literally sleeping). Is there a better way to go about this?